src/main/src/main/java/org/geoserver/catalog/impl/AbstractCatalogFacade.java

```diff
@@ -160,14 +160,16 @@ private void resolveLayerGroupLayers(List<PublishedInfo> layers) {
 
     protected void resolve(StyleInfo style) {
         setId(style);
+        StyleInfoImpl s = (StyleInfoImpl) style;
+        s.setCatalog(getCatalog());
```

Comment:
Makes some sense but I'm wondering... the catalog field is not in the StyleInfo interface, and I don't see anything using it. So two questions:

* How is this causing a problem?
* Should it be just removed instead?

Reply:
the catalog field is not in the StyleInfo interface, but it's required by `getSLD()` and `getStyle()`. Using a dangling reference to a defuncted Catalog instance is a problem because it's working by accident as explained in the description. I don't know what problems could happen when calling `getSLD()` or `getStyle()` on the temporary catalog that survived the load process but shares the ResourcePool and the CatalogInfoLookUps with the final/real Catalog but a different CatalogFacade, but in any case I think it'd be better to get rid of this inconsistency.

Reply:
Oh by the way IIRC there was a functional defect caused by this when the datadir-catalog-loader created a second temporary copy of the catalog and transferred the contents to the final catalog, but now I've changed it to follow the same workflow as DefaultGeoServerLoader.